File: 2138-divide-a-string-into-groups-of-size-k/2138-divide-a-string-into-groups-of-size-k.py
class Solution:
    def divideString(self, s: str, k: int, fill: str) -> List[str]:
        result = []
        n = len(s)
        if n % k != 0:
            count =(int(math.ceil(n / k)) * k) - n
            s += fill * count
        for i in range(0, n, k):
            result.append(s[i:i+k])
        return result

File: 2138-divide-a-string-into-groups-of-size-k/2138-divide-a-string-into-groups-of-size-k.java
class Solution {
    public String[] divideString(String s, int k, char fill) {
        int extra, groups, n=s.length();
        groups=(int)Math.ceil((double)n/k);
        String result[]=new String[groups];
        extra=(groups*k)-n;
        for(int i=1; i<=extra; i++)
            s+=fill;
        for(int i=0, j=0; i<n; i+=k, j++)
            result[j]=s.substring(i, i+k);
        return result;
    }
}